ambry-clustermap/src/main/java/com/github/ambry/clustermap/AmbryReplica.java

```diff
@@ -79,10 +80,19 @@ public boolean isSealed() {
     return isSealed;
   }
 
+  @Override
```

Comment:
isSealed and isPartiallySealed has a little bit overlap.
When isSealed, is isPartiallySealed true or false? They are mutual exclusive or has some overlap?

Reply:
If they are mutually exclusive, when we set isPartiallySealed, we may set the opposite to isSealed. Vise verse. Otherwise, we will have some inconsistency between the variables.

Is one variable with enum type better,
NOT_SEALED,
PARTIALLY_SEALED,
SEALED.

Reply:
I agree that enum names being NOT_SEALED, PARTIALLY_SEALED and SEALED would be better. But we are so used to calling partition state ReadOnly and ReadWrite that I don't want to change it.

Reply:
To answer the first question -- there is definitely overlap between `isSealed` and `isPartiallySealed`. For example, every replica that is sealed is by definition also partially sealed. But in the code, for simplicity, we are treating it as separate states to avoid confusion.

Reply:
It'll cause some trouble if we don't handle the "overlap" part correctly.
For example, when we call "setSealedState" to set isSealed to true, should we call "setPartiallySealedState" to set isPartiallySealed to true.

Internally we should have only one single variable which is an enum. But we can keep the interface/functions not changed.

Reply:
Yeah, I am also wondering if having one internal variable would be better for reasons that Jing mentioned